skyportal/handlers/api/source.py

```diff
@@ -1017,31 +1056,33 @@ class Validator(Schema):
                     readable_classifications_json.append(classification_dict)
 
                 obj_list[-1]["classifications"] = readable_classifications_json
-                obj_list[-1]["annotations"] = sorted(
-                    Annotation.query_records_accessible_by(self.current_user).filter(
-                        Annotation.obj_id == obj.id
-                    ),
-                    key=lambda x: x.origin,
-                )
-                obj_list[-1]["last_detected_at"] = obj.last_detected_at(
-                    self.current_user
-                )
-                obj_list[-1]["last_detected_mag"] = obj.last_detected_mag(
-                    self.current_user
-                )
-                obj_list[-1]["peak_detected_at"] = obj.peak_detected_at(
-                    self.current_user
-                )
-                obj_list[-1]["peak_detected_mag"] = obj.peak_detected_mag(
-                    self.current_user
-                )
-                obj_list[-1]["gal_lon"] = obj.gal_lon_deg
-                obj_list[-1]["gal_lat"] = obj.gal_lat_deg
-                obj_list[-1]["luminosity_distance"] = obj.luminosity_distance
-                obj_list[-1]["dm"] = obj.dm
-                obj_list[-1][
-                    "angular_diameter_distance"
-                ] = obj.angular_diameter_distance
+
+                if not remove_nested:
+                    obj_list[-1]["annotations"] = sorted(
+                        Annotation.query_records_accessible_by(
+                            self.current_user
+                        ).filter(Annotation.obj_id == obj.id),
+                        key=lambda x: x.origin,
+                    )
+                    obj_list[-1]["last_detected_at"] = obj.last_detected_at(
+                        self.current_user
+                    )
+                    obj_list[-1]["last_detected_mag"] = obj.last_detected_mag(
+                        self.current_user
+                    )
+                    obj_list[-1]["peak_detected_at"] = obj.peak_detected_at(
+                        self.current_user
+                    )
+                    obj_list[-1]["peak_detected_mag"] = obj.peak_detected_mag(
+                        self.current_user
+                    )
+                    obj_list[-1]["gal_lon"] = obj.gal_lon_deg
+                    obj_list[-1]["gal_lat"] = obj.gal_lat_deg
+                    obj_list[-1]["luminosity_distance"] = obj.luminosity_distance
+                    obj_list[-1]["dm"] = obj.dm
+                    obj_list[-1][
+                        "angular_diameter_distance"
+                    ] = obj.angular_diameter_distance
```

Comment:
This whole block needs to be de-indented (not nested data)

Reply:
Done